cider/exec/nextgen/jitlib/base/JITValue.h

```diff
@@ -53,6 +53,8 @@ class JITValuePointer {
 
   operator JITValue&() { return *ptr_; }
 
+  JITValuePointer operator[](JITValue& index);
```

Comment:
I am confusing why need this and why return type is still a `JITValuePointer`.
any scenario?

Reply:
Yes, ```JITValuePointer``` is a little confusing. All JIT relate variables are all ```JITValuePointer```, holds the memory and builds operations. This operation means getting element at index from a ```JITValue``` with POINTER type and return it as a ```JITValuePointer``` with ```sub_type``` type.